### PR TITLE
feat(spanner): add samples for custom timeout and retry settings

### PR DESCRIPTION
**Pull request by larkee**
This sample shows how to set custom timeout and retry settings on client calls.

### CODE REVIEW

**Review by bshaffer — APPROVED** [review]

File: spanner/spanner.php [hunk 1]
```diff
@@ -387,6 +387,18 @@
     })
 );
 
+// Set custom timeout and retry
+$application->add((new Command('set_custom_timeout_and_retry'))
```
Comment (bshaffer) [inline spanner/spanner.php:391]: The name should be `set-custom-timeout-and-retry`. Tests are failing as a result.

**Review by bshaffer — CHANGES_REQUESTED** [review]

**Comment by larkee** [comment]
@jdpedrie Could you please confirm this is the correct way to set custom timeout and retry settings for a method call?

**Review by bshaffer — COMMENTED** [review]
@larkee This code will not work. The available options are `requestTimeout`, `retries`, and `grpcOptions`. The `retrySettings` you specified can be passed as an element of `grpcOptions`. So it should look like this:

```php
    $database->runTransaction(function (Transaction $t) use ($spanner) {
        $rowCount = $t->executeUpdate(
            "INSERT Singers (SingerId, FirstName, LastName) "
            . " VALUES (10, 'Virginia', 'Watson')",
            [
                'grpcOptions' => [
                    'retrySettings' => [
                        'initialRetryDelayMillis' => 500,
                        'retryDelayMultiplier' => 1.5,
                        'maxRetryDelayMillis' => 64000,
                        'retryableCodes' => [ApiStatus::DEADLINE_EXCEEDED, ApiStatus::UNAVAILABLE],
                    ],
                ],
                'requestTimeout' => 60000,
            ]);
        $t->commit();
        printf('Inserted %d row(s).' . PHP_EOL, $rowCount);
    });
```

Also the tests need to find a way to test this!

**Comment by bshaffer** [comment]
@larkee any update?

**Pull request closed without merging**